src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -282,7 +297,7 @@ public SAMFileWriter makeSAMWriter(final SAMFileHeader header, final boolean pre
      */
 
     public SAMFileWriter makeBAMWriter(final SAMFileHeader header, final boolean presorted, final OutputStream stream) {
-        return initWriter(header, presorted, true, new BAMFileWriter(stream, null, this.getCompressionLevel()));
+        return initWriter(header, presorted, true, new BAMFileWriter(stream, null, this.getCompressionLevel(), this.deflaterFactory));
```

Comment:
Why does this overload of `makeBAMWriter()` go through a completely different code path than the other overload that takes an additional `compressionLevel`?